In [3]:
import pandas as pd

# Instructions

1. Head back to the [data sets page](https://github.com/michaelschung/bc-data-processing/tree/master/datasets), and pick one data set that you find particularly interesting.
2. Download the file and use Pandas to load that data set. Before moving on, make sure that it downloaded as a `.csv` file! Check the assignment posting on Classroom if you're having issues.
3. Explore the data by querying the DataFrame. Use this document to record your explorations according to the specifications below.

_Note: There are tons of free data sets out there for you to explore, and in the future you'll be able to use any that you like. For this assignment, however, pick one of the data sets that I've provided for you._

---

### Recording Explorations

Make at least three _thoughtful queries_ into the DataFrame. A thoughtful query consists of:

- A concrete question (in English) that you're trying to answer
- The query itself (in Python, using Pandas)
- 2-3 sentences (in English) that use the query results to answer the initial question

_Note: these instructions are intentionally vague. It's up to you to take this as seriously as you want, but I strongly encourage you to dig deep and learn something from the data you choose to look at. If you think you've asked all possible interesting questions, ask one more._

# Your "Thoughtful Queries"

In [5]:
# Load your CSV file here, and preview the first 10-15 rows
netflix_titles = pd.read_csv('netflix_titles.csv', index_col = 2)

netflix_titles.head(15)

show_id     type  \
title                                                               
Norm of the North: King Sized Adventure         81145628    Movie   
Jandino: Whatever it Takes                      80117401    Movie   
Transformers Prime                              70234439  TV Show   
Transformers: Robots in Disguise                80058654  TV Show   
#realityhigh                                    80125979    Movie   
Apaches                                         80163890  TV Show   
Automata                                        70304989    Movie   
Fabrizio Copano: Solo pienso en mi              80164077    Movie   
Fire Chasers                                    80117902  TV Show   
Good People                                     70304990    Movie   
Joaquín Reyes: Una y no más                     80169755    Movie   
Kidnapping Mr. Heineken                         70299204    Movie   
Krish Trish and Baltiboy                        80182480    Movie   
Krish Trish and Baltiboy: Battle of Wits        80182483    Movie   
Krish Trish and Baltiboy: Best Friends Forever  80182596    Movie   

                                                                       director  \
title                                                                             
Norm of the North: King Sized Adventure                Richard Finn, Tim Maltby   
Jandino: Whatever it Takes                                                  NaN   
Transformers Prime                                                          NaN   
Transformers: Robots in Disguise                                            NaN   
#realityhigh                                                   Fernando Lebrija   
Apaches                                                                     NaN   
Automata                                                            Gabe Ibáñez   
Fabrizio Copano: Solo pienso en mi              Rodrigo Toro, Francisco Schultz   
Fire Chasers                                                                NaN   
Good People                                                   Henrik Ruben Genz   
Joaquín Reyes: Una y no más                               José Miguel Contreras   
Kidnapping Mr. Heineken                                        Daniel Alfredson   
Krish Trish and Baltiboy                                                    NaN   
Krish Trish and Baltiboy: Battle of Wits            Munjal Shroff, Tilak Shetty   
Krish Trish and Baltiboy: Best Friends Forever      Munjal Shroff, Tilak Shetty   

                                                                                             cast  \
title                                                                                               
Norm of the North: King Sized Adventure         Alan Marriott, Andrew Toth, Brian Dobson, Cole...   
Jandino: Whatever it Takes                                                       Jandino Asporaat   
Transformers Prime                              Peter Cullen, Sumalee Montano, Frank Welker, J...   
Transformers: Robots in Disguise                Will Friedle, Darren Criss, Constance Zimmer, ...   
#realityhigh                                    Nesta Cooper, Kate Walsh, John Michael Higgins...   
Apaches                                         Alberto Ammann, Eloy Azorín, Verónica Echegui,...   
Automata                                        Antonio Banderas, Dylan McDermott, Melanie Gri...   
Fabrizio Copano: Solo pienso en mi                                                Fabrizio Copano   
Fire Chasers                                                                                  NaN   
Good People                                     James Franco, Kate Hudson, Tom Wilkinson, Omar...   
Joaquín Reyes: Una y no más                                                         Joaquín Reyes   
Kidnapping Mr. Heineken                         Jim Sturgess, Sam Worthington, Ryan Kwanten, A...   
Krish Trish and Baltiboy                        D

## Thoughtful Query #1

Question: How many movies made in the United States that are rated TV-MA were released between 2010 and 2019 (inclusive)?

In [6]:
# Query 1 here
netflix_titles[(netflix_titles['country'] == 'United States') & (netflix_titles['rating'] == 'TV-MA') & (netflix_titles['release_year'] >= 2010) & (netflix_titles['release_year'] <= 2019)]

show_id     type  \
title                                                                 
Fire Chasers                                      80117902  TV Show   
Marc Maron: Too Real                              80177405    Movie   
Christiane Amanpour: Sex & Love Around the World  81011682  TV Show   
Gerald's Game                                     80128722    Movie   
Hold the Dark                                     80157072    Movie   
...                                                    ...      ...   
The Get Down                                      80025601  TV Show   
Ash vs. Evil Dead                                 80049277  TV Show   
Chelsea                                           80049872  TV Show   
Weeds                                             70136122  TV Show   
Maron                                             70286564  TV Show   

                                                         director  \
title                                                               
Fire Chasers                                                  NaN   
Marc Maron: Too Real                                 Lynn Shelton   
Christiane Amanpour: Sex & Love Around the World              NaN   
Gerald's Game                                       Mike Flanagan   
Hold the Dark                                     Jeremy Saulnier   
...                                                           ...   
The Get Down                                                  NaN   
Ash vs. Evil Dead                                             NaN   
Chelsea                                                       NaN   
Weeds                                                         NaN   
Maron                                                         NaN   

                                                                                               cast  \
title                                                                                                 
Fire Chasers                                                                                    NaN   
Marc Maron: Too Real                                                                     Marc Maron   
Christiane Amanpour: Sex & Love Around the World                                                NaN   
Gerald's Game                                     Carla Gugino, Bruce Greenwood, Henry Thomas, C...   
Hold the Dark                                     Jeffrey Wright, Alexander Skarsgård, James Bad...   
...                                                                                             ...   
The Get Down                                      Justice Smith, Shameik Moore, Herizen Guardiol...   
Ash vs. Evil Dead                                 Bruce Campbell, Lucy Lawless, Ray Santiago, Da...   
Chelsea                                                                                         NaN   
Weeds                                             Mary-Louise Parker, Hunter Parrish, Alexander ...   
Maron                                             Marc Maron, Judd Hirsch, Josh Brener, Nora Zeh...   

                                                        country  \
title                                                             
Fire Chasers                                      United States   
Marc Maron: Too Real                              United States   
Christiane Amanpour: Sex & Love Around the World  United States   
Gerald's Game                                     United States   
Hold the Dark                                     United States   
...                                                         ...   
The Get Down                                      United States   
Ash vs. Evil Dead                                 United States   
Chelsea                                           United States   
Weeds                                             United States   
Maron                                             United States   

                 

Insight: The table now only displays all the shows that were made in the United States with a rating of TV-MA and released between the years of 2010 and 2019. Although not all of the rows are displayed, at the bottom of the table, it's written "601 rows" which means there are 601 shows altogether that fit this description. 

## Thoughtful Query #2

Question: Which rating is most common for movies out of the following: TV-MA, TV-Y7, TV-Y, R, TV-14, G, TV-G, PG, TV-PG, PG-13, NC-17?

In [7]:
# Query 2 here
tv_ma = len(netflix_titles[(netflix_titles['type'] == 'Movie') & (netflix_titles['rating'] == 'TV-MA')])
tv_y7 = len(netflix_titles[(netflix_titles['type'] == 'Movie') & (netflix_titles['rating'] == 'TV-Y7')])
tv_y = len(netflix_titles[(netflix_titles['type'] == 'Movie') & (netflix_titles['rating'] == 'TV-Y')])
r = len(netflix_titles[(netflix_titles['type'] == 'Movie') & (netflix_titles['rating'] == 'R')])
tv_14 = len(netflix_titles[(netflix_titles['type'] == 'Movie') & (netflix_titles['rating'] == 'TV-14')])
g = len(netflix_titles[(netflix_titles['type'] == 'Movie') & (netflix_titles['rating'] == 'G')])
tv_g = len(netflix_titles[(netflix_titles['type'] == 'Movie') & (netflix_titles['rating'] == 'TV-G')])
pg = len(netflix_titles[(netflix_titles['type'] == 'Movie') & (netflix_titles['rating'] == 'PG')])
tv_pg = len(netflix_titles[(netflix_titles['type'] == 'Movie') & (netflix_titles['rating'] == 'TV-PG')])
pg_13 = len(netflix_titles[(netflix_titles['type'] == 'Movie') & (netflix_titles['rating'] == 'PG-13')])
nc_17 = len(netflix_titles[(netflix_titles['type'] == 'Movie') & (netflix_titles['rating'] == 'NC-17')])

print(' TV-MA:', tv_ma, ' TV-Y7:', tv_y7, ' TV-Y:', tv_y, ' R:', r, ' TV-14:', tv_14, ' G:', g, ' TV-G:', tv_g, ' PG:', pg, ' TV-PG:', tv_pg, ' PG-13:', pg_13, ' NC-17:', nc_17)

 TV-MA: 1348  TV-Y7: 69  TV-Y: 41  R: 506  TV-14: 1038  G: 36  TV-G: 80  PG: 183  TV-PG: 432  PG-13: 286  NC-17: 2


Insight: The values at the bottom represent the frequency of each of the ratings of movies that show up in the data set. From these values, we can see that the TV-MA has the highest value of 1348 which means that it is the most common rating for movies on netflix.

## Thoughtful Query #3

Question: Were more movies released between 2000-2010 or 2011-2020? 

In [8]:
# Query 3 here
req_1 = len(netflix_titles[(netflix_titles['release_year'] >= 2000) & (netflix_titles['release_year'] <= 2010) & (netflix_titles['type'] == "Movie")])
req_2 = len(netflix_titles[(netflix_titles['release_year'] >= 2011) & (netflix_titles['release_year'] <= 2020)& (netflix_titles['type'] == "Movie")])

print('# of movies released between:')
print('2000 & 2010:', req_1) 
print('2011 & 2020:',req_2)

# of movies released between:
2000 & 2010: 623
2011 & 2020: 3305


Insight: From the data described above, you can see that between 2000 and 2010, a total of 623 movies were released and between 2011 and 2020, a total of 3305 were released. Based on this, the answer to the question is that more movies were released between 2011 and 2020.